src/main/java/edu/harvard/h2ms/service/NotificationServiceImpl.java

```diff
@@ -0,0 +1,180 @@
+package edu.harvard.h2ms.service;
+
+import edu.harvard.h2ms.domain.core.Notification;
+import edu.harvard.h2ms.domain.core.User;
+import edu.harvard.h2ms.repository.NotificationRepository;
+import edu.harvard.h2ms.repository.UserRepository;
+import java.util.Map;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+@Component
+@Service("notificationService")
+public class NotificationServiceImpl {
+
+  // this determines how long it takes for next notification
+  public enum NotificationFrequency {
+    HALFMINUTE("HALFMINUTE", 30L), // for testing
+    DAILY("DAILY", 86400L),
+    WEEKLY("WEEKLY", 604800L),
+    MONTHLY("MONTHLY", 2592000L),
+    UNDEFINED("UNDEFINED", 0L);
+
+    public final String stringRepresentation;
+    public final long seconds;
+
+    NotificationFrequency(String stringRepresentation, long seconds) {
+      this.stringRepresentation = stringRepresentation;
+      this.seconds = seconds;
+    }
+
+    public static NotificationFrequency getNotificationFrequency(String stringRepresentation) {
+
+      for (NotificationFrequency nf : NotificationFrequency.class.getEnumConstants()) {
+        if (nf.stringRepresentation == stringRepresentation) {
```

Comment:
You can just do: `NotificationFrequency.valueOf(stringRepresentation)` which [will return the right enum](https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html#valueOf(java.lang.Class,%20java.lang.String)).  

But if you're comparing strings like this, .equals should be used instead of ==, otherwise it's not going to work.